src/target/imxrt.c

```diff
@@ -140,10 +175,14 @@ typedef struct imxrt_flexspi_lut_insn {
 
 typedef struct imxrt_priv {
 	imxrt_boot_src_e boot_source;
+	uint16_t chip_id;
+	uint32_t flexspi_root;
 	uint32_t mpu_state;
 	uint32_t flexspi_lut_state;
 	uint16_t flexspi_cached_commands[4];
 	imxrt_flexspi_lut_insn_s flexspi_prg_seq_state[4][8];
+	bool flash_in_package;
+	char variant_string[16];
```

Comment:
Given the length of this is something that can change, it makes sense to us to make the size component a `#define` above - `#define IMXRT_NAME_MAX_LENGTH 16U` for example - so it's more trivial to adjust. This then gives you a define to use in the snprintf() below rather than having to rely on sizeof() doing something sensible on all possible compilation platforms.

We would also suggest this is the part's name, so `variant_string` -> `name`? It'd make it make more sense below in our opinion.